eth/api.go

```diff
@@ -478,7 +478,9 @@ func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, txHash common.
 		deadlineCtx, cancel := context.WithTimeout(ctx, timeout)
 		go func() {
 			<-deadlineCtx.Done()
-			tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			if deadlineCtx.Err() == context.DeadlineExceeded {
+				tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			}
```

Comment:
The VM doesn't need to be interrupted if `context.Canceled` is called, because that means the VM already finished executing. It's not a big deal, but I didn't see any point in sending an error interrupt to a VM that's already successfully concluded.